backoffice/src/pages/products/components/CreateProduct.js

```diff
@@ -114,23 +114,27 @@ class CreateProduct extends Component {
     categoryId,
     variants,
     imageUrl,
-  }) {
-    this.setState({
-      name,
-      description,
-      available,
-      displayPrice,
-      brandId,
-      selectedOptions,
-      attributes,
-      productId,
-      categoryId,
-      variants,
-      imageUrl,
-      selectedOptionsValues: this.computeSelectedOptionsValues(selectedOptions, variants),
-    });
-  }
+  }=this.props;
+this.state = {...this.state,
+	name,
+    description,
+    available,
+    brandId,
+    selectedOptions,
+    attributes,
+    productId,
+    displayPrice,
+    categoryId,
+    variants,
+    imageUrl,
+    selectedOptionsValues: this.computeSelectedOptionsValues(selectedOptions, variants)}
+}
 
+ componentWillReceiveProps(props){
+ //see https://github.com/apollographql/react-apollo/issues/1611 the problem inherent with componentWillReceiveProps and react-apollo 2
+ console.log("TestWillReceiveProps",props)
```

Comment:
This `console.log` should be removed, as well as the `componentWillReceiveProps` event if it's not used anymore.

If you wish to keep the comment, then put it above `componentWillMount()`

Reply:
i kept this just to show you the bug after the second click on edit product TestWillReceiveProps will never be showed in console. you can remove the componentWillReceiveProps